propagate_const.h

```diff
@@ -41,84 +67,107 @@ class propagate_const {
 
  public:
   // [propagate_const.ctor], constructors
-  constexpr propagate_const() = default;
+  PROPAGATE_CONST_CONSTEXPR propagate_const() = default;
 
   propagate_const(const propagate_const& p) = delete;
 
-  constexpr propagate_const(propagate_const&& p) = default;
+  PROPAGATE_CONST_CONSTEXPR propagate_const(propagate_const&& p) = default;
 
+#ifdef PROPAGATE_CONST_HAS_EXPRESSION_SFINAE
```

Comment:
IMO, there is a missing constructor, with the EXPLICIT variant

```c++
EXPLICIT constexpr propagate_const(propagate_const<U> const& pu);
```

Reply:
Copy constructors are deleted for propagate const as it lets one get round const-propagation.

Reply:
yes, I forgot about that.